arm9/src/timer3.c

```diff
@@ -0,0 +1,67 @@
+#include "timer3.h"
+
+struct Timer3Data timer3_data;
+
+
+THUMB_FUNC void Init_Timer3()
+{
+    timer3_data.Timer3Counter = 0;
+    timer3_data.NeedReset = FALSE;
+
+    reg_OS_TM3CNT_H = 0;
+    reg_OS_TM3CNT_L = 0;
+    reg_OS_TM3CNT_H = 0xc1; // start timer3 with f/64 and irq enable
+
+    OS_SetIrqFunction(0x40, &CountUpTimer3);
+    OS_EnableIrqMask(0x40); // irq on timer3 overflow
+}
+
+
+THUMB_FUNC void CountUpTimer3()
+{
+    timer3_data.Timer3Counter++;
+
+    if (timer3_data.NeedReset)
+    {
+        reg_OS_TM3CNT_H = 0;
+        reg_OS_TM3CNT_L = 0;
+        reg_OS_TM3CNT_H = 0xc1;
+        timer3_data.NeedReset = FALSE;
+    }
+    u32 *ptr = (u32 *)0x027e0000;
+    u32 offset = 0xffe;
+    ptr[offset] |= 0x40;
```

Comment:
this should be an address in DTCM IIRC, if so please change to reference that correctly

Reply:
This was actually `HW_INTR_CHECK_BUF`, thanks